backend/app/src/main/java/com/moh/phlat/backend/esb/json/MaintainHdsRequest.java

```diff
@@ -73,30 +73,102 @@ private Map<String,Object> createEsbMetadata() {
 		providerDetails.setAddresses(createAddressDtos());
 		providerDetails.setTelecommunication(createTelecomunicationDtos());
 		providerDetails.setElectronicAddresses(createElectronicAddressDtos());
 		return providerDetails;
 	}
 	
-	private FacilityDto createFacilityDto() {
-		FacilityDto facility = new FacilityDto();
-		return facility;
+	private List<OrgNameDto> createOrgNameDtos() {
+		List<OrgNameDto> orgNameList = new ArrayList<>();
+		
+		OrgNameDto orgName = new OrgNameDto();
+		orgName.setName(data.getHdsName());
+		if (data.getHdsPreferredNameFlag() != null) {
+			orgName.setPreferred(Boolean.parseBoolean(data.getHdsPreferredNameFlag()));
+		}
+		orgName.setTypeCode("CURR");
+		orgName.setEffectiveStartDate(data.getCreatedAt());
+		orgNameList.add(orgName);
+		
+		return orgNameList;
 	}
 	
 	private HdsTypeDto createHdsTypeDto() {
 		HdsTypeDto hdsType = new HdsTypeDto();
+		hdsType.setHdsType(data.getHdsType());
+		hdsType.setEffectiveStartDate(data.getCreatedAt());
 		
 		return hdsType;
 	}
 	
+	private List<CollegeIdentifierDto> createIdentifierDtos() {
+		
+		List<CollegeIdentifierDto> identifierList = new ArrayList<>();
+		
+		if (StringUtils.hasText(data.getHdsProviderIdentifier1())
+				&& StringUtils.hasText(data.getHdsProviderIdentifierType1())) {
+			CollegeIdentifierDto identifier1 = new CollegeIdentifierDto();
+			identifier1.setIdentifier(data.getHdsProviderIdentifier1());
+			identifier1.setType(data.getHdsProviderIdentifierType1());
+			identifier1.setEffectiveStartDate(data.getCreatedAt());
+			identifierList.add(identifier1);
+		}
+		if (StringUtils.hasText(data.getHdsProviderIdentifier2())
+				&& StringUtils.hasText(data.getHdsProviderIdentifierType2())) {
+			CollegeIdentifierDto identifier2 = new CollegeIdentifierDto();
+			identifier2.setIdentifier(data.getHdsProviderIdentifier2());
+			identifier2.setType(data.getHdsProviderIdentifierType2());
+			identifier2.setEffectiveStartDate(data.getCreatedAt());
+			identifierList.add(identifier2);
+		}
+		if (StringUtils.hasText(data.getHdsProviderIdentifier3())
+				&& StringUtils.hasText(data.getHdsProviderIdentifierType3())) {
+			CollegeIdentifierDto identifier3 = new CollegeIdentifierDto();
+			identifier3.setIdentifier(data.getHdsProviderIdentifier3());
+			identifier3.setType(data.getHdsProviderIdentifierType3());
+			identifier3.setEffectiveStartDate(data.getCreatedAt());
+			identifierList.add(identifier3);
+		}
+		
+		if (identifierList.isEmpty()) {
+			return null;
+		}
+		return identifierList;
+	}
+	
+	private JurisdictionNameCodeDto createJurisdictionDto() {
+		JurisdictionNameCodeDto jursidictionDto = new JurisdictionNameCodeDto();
+		jursidictionDto.setJurisdicationNameCode(data.getPhysicalProvince());
+		
+		return jursidictionDto;
+	}
+	
+	private List<StatusDto> createStatusDtos() {
+		List<StatusDto> statusList = new ArrayList<>();
+		
+		StatusDto status = new StatusDto();
+		status.setTypeCode("ACTIVE");
+		status.setClassCode("AE");
```

Comment:
Just noticed some of the fields are hardcoded. I am not sure if they need to be pulled from the request. Like status and setCommunicationPurposeCode("BC")

Reply:
PLR sent me errors saying I needed to fill out these fields, so I hardcoded them based off of the options I saw in the database CT tables; we don't have these fields in the pharmacies file we're loading from, and I doubt we'll ever have them in there or obtained from any other source. We also know that these are strictly orgs/HDSs that we're loading, so I'm not too concerned about these hardcodings for now.